crypto/src/main/java/org/cryptimeleon/incentive/crypto/Util.java

```diff
@@ -19,24 +17,68 @@
 
     /**
      * Hash function to retrieve ZnElement gamma in spend-deduct
+     * TODO think about putting additional input to replace k
+     * TODO no, we have the new commitments in the hash
      *
      * @param zn    ZnElement to retrieve
-     * @param K     earn vector to hash
-     * @param dsid  disid to hash
+     * @param dsid  dsid to hash
      * @param tid   tid to hash
      * @param cPre0 cPre0 to hash
      * @param cPre1 cPre1 to hash
      * @return hashed ZnElement gamma
      */
-    public static Zn.ZnElement hashGamma(Zn zn, RingElementVector K, GroupElement dsid, Zn.ZnElement tid, GroupElement cPre0, GroupElement cPre1) {
+    public static Zn.ZnElement hashGamma(Zn zn, GroupElement dsid, Zn.ZnElement tid, GroupElement cPre0, GroupElement cPre1) {
         var hashfunction = new HashIntoZn(zn);
         var accumulator = new ByteArrayAccumulator();
-        K.stream().forEachOrdered(k -> accumulator.escapeAndSeparate(k.getUniqueByteRepresentation()));
         accumulator.escapeAndSeparate(dsid.getUniqueByteRepresentation());
         accumulator.escapeAndSeparate(tid.getUniqueByteRepresentation());
         accumulator.escapeAndSeparate(cPre0.getUniqueByteRepresentation());
         accumulator.escapeAndSeparate(cPre1.getUniqueByteRepresentation());
         return hashfunction.hash(accumulator.extractBytes());
     }
 
+    /**
+     * Returns a Vector of size many null elements.
+     *
+     * @param size size of the vector
+     * @return vector of type BigInteger
+     */
+    public static Vector<BigInteger> getNullBigIntegerVector(int size) {
+        return Vector.iterate(null, e -> e, size);
+    }
+
+    /**
+     * Returns a Vector of size many zero elements.
+     *
+     * @param size size of the vector
+     * @return vector of type BigInteger
+     */
+    public static Vector<BigInteger> getZeroBigIntegerVector(int size) {
+        return Vector.iterate(BigInteger.ZERO, e -> e, size);
+    }
+
+    /**
+     * Returns a Vector of size many one elements.
+     *
+     * @param size size of the vector
+     * @return vector of type BigInteger
+     */
+    public static Vector<BigInteger> getOneBigIntegerVector(int size) {
+        return Vector.iterate(BigInteger.ONE, e -> e, size);
+    }
+
+    /**
+     * Determines whether {@literal lowerLimits[i] <= pointsVector[i] <= upperLimits[i] for all i}.
+     * If limits are null, relation skipped.
+     *
+     * @param pointsVector a vector of points
+     * @param lowerLimits  a vector of lower limits
+     * @param upperLimits  a vector of upper limits
+     * @return whether are relations are satisfied
+     */
+    public static boolean arePointsInRange(RingElementVector pointsVector, Vector<BigInteger> lowerLimits, Vector<BigInteger> upperLimits) {
```

Comment:
Would prefer naming like "pointsInRange".
But maybe this is just personal preference.

Reply:
Well I guess it is my preference to include the are to signal that the function returns a boolean and not points :)